pkg/test/step.go

```diff
@@ -402,13 +417,18 @@ func (s *Step) Run(namespace string) []error {
 		return testErrors
 	}
 
-	for i := 0; i < s.GetTimeout(); i++ {
-		testErrors = s.Check(namespace)
+	timeoutF := float64(s.GetTimeout())
+	start := time.Now()
+
+	for elapsed := 0.0; elapsed < timeoutF; elapsed = time.Since(start).Seconds() {
```

Comment:
I need to take a closer look at this, I'll circle back to this on Monday. I would expect we have a context set with that timeout and then we pass context around also including all the methods.

Maybe this is valid implementation as well and we can do the context one as follow-up though.

Reply:
I think that would be better, and that is the direction I was going (that is why I added ctx to many methods). But also i was trying not to do too much code changes to try to adhere to @kensipe guidelines. My reasoning was: if it's a good practice and it has minimal  or no impact, the do it. (That is why adding the context fell in).

Reply:
one of the timeouts... should be passed thru with context... completely agree with @alenkacz .   not holding this PR to that... but is the right way to manage 1 or 2 of the timeouts.  a topic which should tackle next if possible.